webrender/src/profiler.rs

```diff
@@ -352,6 +359,88 @@ impl ProfileGraph {
     }
 }
 
+struct GpuFrame {
+    total_time: u64,
+    samples: Vec<GpuSample<GpuProfileTag>>,
+}
+
+struct GpuFrameCollection {
+    frames: VecDeque<GpuFrame>,
+}
+
+impl GpuFrameCollection {
+    fn new() -> GpuFrameCollection {
+        GpuFrameCollection {
+            frames: VecDeque::new(),
+        }
+    }
+
+    fn push(&mut self, total_time: u64, samples: Vec<GpuSample<GpuProfileTag>>) {
+        if self.frames.len() == 20 {
+            self.frames.pop_back();
+        }
+        self.frames.push_front(GpuFrame {
+            total_time: total_time,
+            samples: samples,
+        });
+    }
+}
+
+impl GpuFrameCollection {
+    fn draw(&self,
+            x: f32,
+            y: f32,
+            debug_renderer: &mut DebugRenderer) -> Rect<f32> {
+        let graph_size = Size2D::new(1024.0, 320.0);
+        let bounding_rect = Rect::new(Point2D::new(x, y),
+                                      Size2D::new(graph_size.width + 16.0,
+                                                  graph_size.height + 16.0));
+        let graph_rect = bounding_rect.inflate(-8.0, -8.0);
+
+        debug_renderer.add_quad(bounding_rect.origin.x,
+                                bounding_rect.origin.y,
+                                bounding_rect.origin.x + bounding_rect.size.width,
+                                bounding_rect.origin.y + bounding_rect.size.height,
+                                &ColorF::new(0.1, 0.1, 0.1, 0.8),
+                                &ColorF::new(0.2, 0.2, 0.2, 0.8));
+
+        let w = graph_rect.size.width;
+        let mut y0 = graph_rect.origin.y;
+
+        let mut max_time = 0;
+        for frame in &self.frames {
+            max_time = cmp::max(max_time, frame.total_time);
```

Comment:
There's a `max_by_key` method for this: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max_by_key